linked_list/linked_list.py

```diff
@@ -21,15 +21,48 @@ class LinkedList:
   def __init__(self):
     self.head = None
     self.tail = None
+    self.max = None
 
   def add_to_tail(self, value):
-    pass
+    # Create new node
+    node = Node(value)
+    # If the ll is not empty
+    if self.tail is not None:
+      # Then set the tail's next to the new node
+      self.tail.set_next(node)
+      # Set max to node if higher than current value
+      if self.max.value < node.value:
+        self.max = node
+    else:
+      # If it is empty, set the new node to the head
+      self.head = node
+      # And set it to max
+      self.max = node
 
   def remove_head(self):
-    pass
+    # Check if the head is None
+    if self.head is not None:
+      # set the head nodes next node value to a temp var
+      new_head = self.head.next_node
+      # delete the head node (not actually needed because python deallocates on its own if no more refrences)
+      del(self.head)
+      # Then set head to that temp
+      self.head = new_head
 
   def contains(self, value):
-    pass
+    # Set the current node to the head
+    curr_node = self.head
+    
+    while True:
```

Comment:
Would would be a better condition to check here. I know this passes tests but I think it's possible to use this function as intended but still get an infinite loop in some cases.

Reply:
What would be a better condition to check? How it is currently, there can't be an infinite loop since there are the two checks at the top of the loop and current will always be updated to the next node and which will eventually be None